syntax/javascript.vim

```diff
@@ -337,7 +337,7 @@ if version >= 508 || !exists("did_javascript_syn_inits")
   HiLink jsSwitchColon          Noise
   HiLink jsClassMethodType      Type
   HiLink jsObjectMethodType     Type
-  HiLink jsClassDefinition      jsFuncName
+  HiLink jsClassDefinition      Constant
```

Comment:
Yeah I did read it. Think is, there's intention and use, and then there's syntax accuracy. The methodology we try to take here (and don't always do it perfectly, mind you) is to match it accurately, then leave it up to each particular dev to customize as appropriate. Therefore, we match it with function because that's what it is, but allow the flexibility for it to be changed by a dev later.